src/Symfony/Component/Form/Extension/DataCollector/FormDataExtractor.php

```diff
@@ -24,16 +29,18 @@
 class FormDataExtractor implements FormDataExtractorInterface
 {
     /**
-     * @var ValueExporter
+     * @var VarCloner
      */
-    private $valueExporter;
+    private $varCloner;
 
     /**
      * Constructs a new data extractor.
      */
-    public function __construct(ValueExporter $valueExporter = null)
+    public function __construct(ValueExporter $valueExporter = null, $triggerDeprecationNotice = true)
     {
-        $this->valueExporter = $valueExporter ?: new ValueExporter();
+        if (null !== $valueExporter && $triggerDeprecationNotice) {
```

Comment:
in which case do we need to avoid the deprecation ? The ValueExporter is ignored now, so we could stop injecting it in the core.